### PR TITLE
:mage: Render new nodetypes for Sphinx autodoc

### PR DESCRIPTION
**Pull request by agoose77**
This PR adds some basic styling for mdast nodes emitted by https://github.com/executablebooks/sphinx-ext-mystmd

### CODE REVIEW

File: packages/myst-to-react/src/autodoc.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+};
+
+const preParentRenderer = ({ node }: { node: HasTextChildren }) => {
+  const [child] = node.children;
+  return (
+    <span
+      style={{
+        whiteSpace: 'pre',
+      }}
+    >
```
Comment (rowanc1) [inline packages/myst-to-react/src/autodoc.tsx:22]:
```suggestion
    <span className="whitespace-pre">
```

File: packages/myst-to-react/src/autodoc.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  },
+  descName({ node }: { node: HasTextChildren }) {
+    const [child] = node.children;
+    return (
+      <span
+        style={{
+          fontWeight: 'bold',
+          whiteSpace: 'pre',
+        }}
+      >
```
Comment (rowanc1) [inline packages/myst-to-react/src/autodoc.tsx:55]:
```suggestion
      <span className="whitespace-pre font-bold">
```

File: packages/myst-to-react/src/autodoc.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    );
+  },
+  descSignature({ node }: { node: HasChildren }) {
+    return (
+      <dt
+        style={{
+          fontWeight: 'lighter',
+          fontFamily: "Menlo, Consolas, 'DejaVu Sans Mono', monospace",
+        }}
+      >
```
Comment (rowanc1) [inline packages/myst-to-react/src/autodoc.tsx:102]:
```suggestion
      <dt className="font-mono font-light">
```

File: packages/myst-to-react/src/autodoc.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        <MyST ast={node.children} />
+      </dd>
+    );
+  },
+  descAnnotation: renderChildren,
+  descParameter: renderChildren,
+  descReturns({ node }: { node: HasChildren }) {
+    return (
+      <>
+        <span>&#x2192; </span>
```
Comment (rowanc1) [inline packages/myst-to-react/src/autodoc.tsx:119]: As this is decoration, probably better in css? You can do this in a `:before` node and set the content to an arrow.

**Review by rowanc1 — CHANGES_REQUESTED** [review]
As these are specific for sphinx, are not used in scientific , and the AST might change. I think these should be their own package. Ideally we have them as an export out of `myst-ext-autodoc/react`?

Doing that makes it a bit harder to do the auto reload, but I think it is maybe worth it?

If we did that, then I still think these should come in by default in the themes we have, but other themes can make different choices.

**Comment by rowanc1** [comment]
In conversation today: we should see how far we can get on doing a pre-transform on the AST to use `kind` to differentiate, e.g., a field list, on a `definitionList`.

**Comment by agoose77** [comment]
Closed in favour of #443

**Pull request closed without merging**